### PR TITLE
Small refactor of Custom Resources topic

### PR DESCRIPTION
**Pull request by mistyhacks**
- Adjusts levels
- Removes some obsolete info
- Rewrite of some of the intro material

### CODE REVIEW

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 1]
```diff
@@ -9,40 +9,49 @@ weight: 20
...
+A *custom resource* is an extension of the Kubernetes API that is not necessarily available in a default
+Kubernetes installation. It represents a customization of a particular Kubernetes installation. However,
+many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.
 
 Custom resources can appear and disappear in a running cluster through dynamic registration,
 and cluster admins can update custom resources independently of the cluster itself.
-Once a custom resource is installed, users can create and access its objects with
-[kubectl](/docs/user-guide/kubectl-overview/), just as they do for built-in resources like *pods*.
+Once a custom resource is installed, users can create and access its objects using the
+[kubectl](/docs/user-guide/kubectl-overview/) command, just as they do for built-in resources like
```
Comment (nikhita) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: Instead of saying "using the kubectl command", I think we should say "using kubectl" - because kubectl is the tool and not the command itself.

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 2]
```diff
@@ -9,40 +9,49 @@ weight: 20
...
+
 A [declarative API](/docs/concepts/overview/working-with-objects/kubernetes-objects/#understanding-kubernetes-objects)
 allows you to _declare_ or specify the desired state of your resource and tries
-to match the actual state to this desired state.
-Here, the controller interprets the structured data as a record of the user's
-desired state, and continually takes action to achieve and maintain this state.
+keep the current state of Kubernetes objects in sync with the desired state.
+The controller interprets the structured data as a record of the user's
+desired state, and continually maintains this state (or generates errors if it
+cannot maintain the state).
```
Comment (nikhita) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: I don't think it is completely accurate to say that it "generate errors" because sometimes if it can't maintain the state, the controller will update the "status" to reflect the current state - but not necessarily fail with an error. I think we can remove the last part in the brackets.

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 3]
```diff
@@ -9,40 +9,49 @@ weight: 20
...
 
-A *custom controller* is a controller that users can deploy and update on a running cluster, independently of the cluster's own lifecycle. Custom controllers can work with any kind of resource, but they are especially effective when combined with custom resources. The [Operator](https://coreos.com/blog/introducing-operators.html) pattern is one example of such a combination. It allows developers to encode domain knowledge for specific applications into an extension of the Kubernetes API.
+You can deploy and update a custom controller on a running cluster, independently
+of the cluster's own lifecycle. Custom controllers can work with any kind of resource,
+but they are especially effective when combined with custom resources. The
+[Operator pattern](https://coreos.com/blog/introducing-operators.html) combines a custom
+resource and a custom controller. You can use custom controllers to encode domain knowledge
```
Comment (nikhita) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: Maybe "Operator pattern..combines custom resources and custom controllers" instead of a **_a_** custom resource and **_a_** custom controller? The idea is to make sure that it is not necessary that there is only one custom resource or only one custom controller, but I'm not sure what is the right way to write it. :grimacing:

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 4]
```diff
@@ -126,12 +135,12 @@ This frees you from writing your own API server to handle the custom resource,
...
-Refer to the [Custom Controller example, which uses Custom Resources](https://github.com/kubernetes/sample-controller)
-for a demonstration of how to register a new custom resource, work with instances of your new resource type,
-and setup a controller to handle events.
+Refer to the [custom controller example](https://github.com/kubernetes/sample-controller)
+for an example of how to register a new custom resource, work with instances of your new resource type,
+and use a controller to handle events.
 
 {{< note >}}
-CRD is the successor to the deprecated *ThirdPartyResource* (TPR) API, and is available as of Kubernetes 1.7.
+CRD is the successor to the deprecated *ThirdPartyResource (TPR)* API.
```
Comment (nikhita) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: TPRs were removed in 1.9 (iirc). So they haven't been deprecated for a long time. :)

**Comment by mistyhacks** [comment]
Addressed feedback by amending. PTAL :)

**Comment by Rajakavitha1** [comment]
Hi @mistyhacks looks like there are some review comments from @kbhawkey .
Could you please take a look at it.

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 5]
```diff
@@ -9,40 +9,48 @@ weight: 20
...
-A *resource* is an endpoint in the [Kubernetes API](/docs/reference/using-api/api-overview/) that stores a collection of [API objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/) of a certain kind. For example, the built-in *pods* resource contains a collection of Pod objects.
+A *resource* is an endpoint in the [Kubernetes API](/docs/reference/using-api/api-overview/) that stores a collection of
+[API objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/) of a certain kind. For example, the built-in *pods* resource contains a collection of Pod objects.
 
-A *custom resource* is an extension of the Kubernetes API that is not necessarily available on every
-Kubernetes cluster.
-In other words, it represents a customization of a particular Kubernetes installation.
+A *custom resource* is an extension of the Kubernetes API that is not necessarily available in a default
+Kubernetes installation. It represents a customization of a particular Kubernetes installation. However,
+many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.
```
[the remark this answers is not in the document]
Comment (mistyhacks) [reply content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: I was more going for something like "functionality". I'll try to think how to reword this.
